src/tools/kdb/file.cpp

```diff
@@ -33,6 +33,17 @@ int FileCommand::execute (Cmdline const & cl)
 		throw invalid_argument (cl.arguments[0] + " is not a valid keyname");
 	}
 
+	if (x.getNamespace () == ElektraNamespace::CASCADING)
+	{
+		throw invalid_argument ("Cannot retrieve the file for a cascading key. Use a concrete namespace.");
+	}
+
+	if (x.getNamespace () != ElektraNamespace::SPEC && x.getNamespace () != ElektraNamespace::SYSTEM &&
+	    x.getNamespace () != ElektraNamespace::USER && x.getNamespace () != ElektraNamespace::DIR)
+	{
+		throw invalid_argument ("Can only retrieve file for a persistable namespace: spec:/, system:/, user:/ or dir:/");
+	}
```

Comment:
@markus2330 I made this decision, even though "find the file where `/foo` is stored" is a valid use case.

This is because, we run into trouble when `/foo` does not exist. What namespace should we then use to find the correct backend?

Even the current `master` implementation is broken in this regard:

```sh
# this is fine, if there is no key it's not really clear what should be returned

kdb file /foo
#> /etc/kdb/default.ecf

# but this is a problem, I'd expect to get the file which contains user:/foo

kdb get system:/foo
#> Did not find key 'system:/foo'
kdb set user:/foo 123
#> Create a new key user:/foo with string "123"
kdb file /foo
#> /etc/kdb/default.ecf
```

The best solution is clearly to only allow concrete namespaces that are persistable in `kdb file`. For the use case of "find the file where `/foo` is stored" (which seems not to be covered on `master` anyway) we should add return a new `kdb lookup`:

```sh
kdb lookup /foo
#> user:/foo

kdb file "$(kdb lookup /foo)"
#> /home/user/.config/default.ecf

Reply:
Ok. Only a pity that you describe this here where it will be totally lost. As code comment directly above or as git commit message would have been better for this.

Reply:
I'll add a comment why not all namespaces are supported. But the context of what `master` does right now is pretty useless in the future.

Reply:
Also, I did update the docs for `kdbGet()`  and those now state that
1. Using any namespace other than cascading, `default:/`, `proc:/`, `spec:/`, `dir:/`, `user:/` or `system:/` for the `parentKey` is an error. (check not implemented yet) I listed the supported ones explicitly in case we add more namespaces in future.
2. When using a  cascading, `default:/`, `proc:/` parent, the value will be set to an empty string.